File: IndexedDbHandler/WorkerSyncConnection/WorkerConsoleReader.cs
using Microsoft.JSInterop;

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace JSWrapper.WorkerConsoleConnection
{
    public class WorkerConsoleReader
    {
        private readonly IJSInProcessRuntime runtime;

        public WorkerConsoleReader(IJSInProcessRuntime runtime)
        {
            this.runtime = runtime;
        }

        public async Task InitializeAsync(string baseUrl)
        {
            await runtime.InvokeAsync<IJSInProcessObjectReference>("importLocalScripts", SettingProvider.JSDirPath + "WorkerConsoleReader.js");
            await runtime.InvokeVoidAsync("SetBaseUrl", baseUrl);
        }

        public void ReadInput()
        {
            string resStr = runtime.Invoke<string>("GetInput");
            XhrResult result = JsonSerializer.Deserialize<XhrResult>(resStr);
            Console.WriteLine($"XHR Result:{result}");
        }
    }

    public class XhrResult
    {
        public int Status { get; set; }

        public string Response { get; set; }

        public override string ToString()
        {
            return $"{Status.ToString()} : {Response}";
        }
    }
}

File: IndexedDbHandler/wwwroot/js/WorkerConsoleReader.js
let baseUrl;
const requestUrl = "_content/JSWrapper/Dummy.html"

function SetBaseUrl(url) {
    baseUrl = url;
}

function GetInput() {
    let xhr = new XMLHttpRequest();
    let url = new URL(requestUrl, baseUrl);
    url.searchParams.set("action", "GetInput");
    xhr.open("GET", url, false);
    xhr.send(null);

    return JSON.stringify({ Status: xhr.status, Response: xhr.response });
}